File: src/main/java/io/github/takenoha/towerdefense/paper/TaggedEscrowDrop.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;

/** PDC identity attached to a physical display of a database-owned escrow drop. */
public record TaggedEscrowDrop(UUID eventId, UUID dropId) {
    public TaggedEscrowDrop {
        Objects.requireNonNull(eventId, "eventId");
        Objects.requireNonNull(dropId, "dropId");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/TaggedEscrowDrop.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** PDC identity attached to a physical display of a database-owned escrow drop. */
@JvmRecord
data class TaggedEscrowDrop(
    val eventId: UUID,
    val dropId: UUID,
) {
    init {
        Objects.requireNonNull(eventId, "eventId")
        Objects.requireNonNull(dropId, "dropId")
    }
}
